### ISSUE TITLE
Proposal: synchronize cgroupv1 deprecation announcements

### ISSUE THREAD

**Issue by haircommander** [opencontainers/runtime-spec#1251]
At a breakout session in [Container Plumbing Days](https://events.linuxfoundation.org/container-plumbing-days/), @AkihiroSuda brought up the topic of a timeline to the cgroupv1 deprecation and removal. In discussing, we came up with a handful of ideas around it. I will begin by summarizing them. As a note: "we" in this conversation includes maintainers of OCI runtimes, and higher level container managers.

- Do we announce deprecation earlier than distros and the kernel (similar to how systemd did)?
- What are the steps that projects should take in announcing/dropping support
- Do we announce as a group?

There seemed to be consensus on the idea of having the OCI runtimes and managers make some sort of shared announcement on the intention of dropping cgroupv1 support. It doesn't require consensus on all implementations, but I think it would be most impactful if it was done together. Note: this conversation is tangential but related to a similar conversation happening in [Kubernetes](https://github.com/kubernetes/enhancements/issues/4569). in this conversation, we're focusing on lower levels, but the Kubernetes community could take a different timeline if they choose. 

@AkihiroSuda suggested we open an issue here to announce follow-up conversations about it. @jberkus created a channel in the OCI slack to discuss further, but I'll let this be an entry point into that conversation. If you know of someone who may be interested in such a conversation, please tag them here or invite them to the [slack channel](https://opencontainers.slack.com/archives/C06UTNP2LL9) (folks can join the OCI slack by following this [link](https://communityinviter.com/apps/opencontainers/join-the-oci-community))

**Comment by haircommander** [comment]
non-exhaustive list of folks who may be interested (please cc others who may be)
cri-o: @mrunalp @saschagrunert @sohankukerkar @kwilczynski 
crun: @giuseppe
podman: @rhatdan @baude @mheon @umohnani8

**Comment by AkihiroSuda** [comment]
## Links to relevant discussions
(maintainers: feel free to edit this post)
- Kubernetes: discussion is on-going: https://github.com/kubernetes/enhancements/pull/4572
- systemd has already disabled cgroup v1 by default in systemd v256 (Feb 2024): https://github.com/systemd/systemd/pull/31442
- Podman has already deprecated cgroup v1 in Podman v5 (March 2024): https://issues.redhat.com/browse/RUN-1957

**Comment by AkihiroSuda** [comment]
## cgroup v1 distros
- Ubuntu 20.04 (END OF STANDARD SUPPORT: Apr 2025, END OF UBUNTU PRO SUPPORT: Apr 2030 https://ubuntu.com/about/release-cycle)
- RHEL/Rocky/Alma/Oracle Linux 8 (Maintenance Support 2 ends: May 31, 2029, ELS add-on ends: May 31, 2032 https://access.redhat.com/support/policy/updates/errata)
- SLES 15 (General Support Ends: 31 July 2028, LTSS Ends: 31 July 2031 https://www.suse.com/lifecycle/#suse-linux-enterprise-server-15)



- - -

I'd suggest keeping cgroup v1 until 2029 at least.
Can be 2030+ depending on the feedbacks from users.

**Comment by utam0k** [comment]
👀 Thanks for giving us a forum to discuss this. First a brief comment, I agree with the direction towards v1 deprication as OCI Runtime maintainer and youki developer.
We will at least have to support it longer than Kubernetes.

For users who see this, supporting v1 means not only more development effort, but also that it will continue to be harder and harder to make new features introduced only in v2 (which will increase) available from containers.

**Comment by mheon** [comment]
On the Podman side, we have already announced that cgv1 is deprecated as of our 5.0 upstream release, and we are actively warning users on v1 distros that support will be removed at a later date.

**Comment by kannon92** [comment]
RHEL 9.4 has deprecated cgroup v1 support.

> cgroupsv1 is now deprecated in RHEL 9

> The cgroups is a kernel subsystem used for process tracking, system resource allocation and partitioning. Systemd service manager supports booting in the cgroups v1 mode as well as in cgroups v2 mode. In Red Hat Enterprise Linux 9, the default mode is v2. In Red Hat Enterprise Linux 10, systemd will not support booting in the cgroups v1 mode and only cgroups v2 mode will be available.

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9-beta/html/9.4_release_notes/deprecated-functionality

**Comment by rata** [comment]
Big +1 to keep it working in OCI runtimes and use the "maintenance mode" or even "not even bug fixes" at some point in the future if needed, and we of course add things that only work in cgroups v2.
